src/cli.ts

```diff
@@ -2,38 +2,59 @@
  * @copyright Microsoft Corporation. All rights reserved.
  */
 import yargs from 'yargs/yargs';
+import { validate as uuidValidate } from 'uuid';
 import { run } from '.';
 
-interface Arguments {
+interface IArguments {
   ID: string;
+  beforeDate: string;
   projectPath: string;
   ksFilePath: string;
 }
 
-const argv = yargs(process.argv.slice(2))
-  .usage('ks-killer <KS ID> [options...]')
-  .showHelpOnFail(true, 'Please provide a KS ID')
-  .command(
-    '$0 <ID>',
-    'KS Killer',
-    (yargs) => {
-      yargs
-        .positional('id', {
-          describe: 'ID of the KS to graduate',
-          type: 'string'
-        })
-        .option('project-path', {
-          alias: 'p',
-          type: 'string',
-          describe: 'Target project to clean. Default to cwd.',
-          default: process.cwd()
-        })
-        .option('ks-file-path', {
-          alias: 'k',
-          type: 'string',
-          describe: 'The file path of the KS. This boosts performance.'
-        });
-    },
-    ({ ID, projectPath, ksFilePath }: Arguments) => run(ID, projectPath, ksFilePath)
-  )
+const argv: IArguments = yargs(process.argv.slice(2))
+  .usage('ks-killer [options...]')
+  .showHelpOnFail(true)
+  .option('ID', {
+    alias: ['id'],
+    describe: 'ID of the KS to graduate',
+    type: 'string'
+  })
+  .option('beforeDate', {
+    alias: ['before-date'],
+    type: 'string',
+    describe: 'Graduate ks before the specific date. e.g. MM/DD/YYYY'
+  })
+  .option('projectPath', {
+    alias: ['project-Path', 'p'],
+    type: 'string',
+    describe: 'Target project to clean. Default to cwd.',
+    default: process.cwd()
+  })
+  .option('ksFilePath', {
+    alias: ['ks-File-Path', 'k'],
+    type: 'string',
+    describe: 'The file path of the KS. This boosts performance.'
+  })
+  .conflicts('ID', 'beforeDate')
+  .check((argv) => {
+    if ((argv.ID || argv.beforeDate)) {
+      if (argv.ID && !uuidValidate(argv.ID)) {
+        throw (new Error('Error: please provide a valid ks-ID (uuid)'));
+      }
+      if (argv.beforeDate && isNaN(Date.parse(argv.beforeDate)) {
+        throw (new Error('Error: please provide a valid date (e.g. MM/DD/YYYY)'));
+      }
+      return true;
+    } else if (/^\w{8}-\w{4}-\w{4}-\w{4}-\w{12}$/.test(argv['_'][0]?.toString())) {
```

Comment:
Can we use `uuidValidate` here?
```suggestion
    } else if (uuidValidate(argv['_'][0]?.toString())) {
```